### PR TITLE
DPR full tests

### PR DESCRIPTION
**Pull request by LysandreJik**
Uses the `ModelTesterMixin` for tests.

- Adds support for `inputs_embeds`
- Adds support for `output_attentions`
- Adds support for `output_hidden_states`.

Some things to note:

- The `namedtuple` `DPRReaderOutput` is not used by the `DPRReader` anymore. This is so for two reasons:
    - There can be no optional values in the `namedtuple` before Python 3.7 (we support 3.6+), and both `hidden_states` and `attentions` should be optional. This `namedtuple` is kept so that the type hints in the tokenizer stay relevant. 
    - With Sylvain's https://github.com/huggingface/transformers/pull/5438, the outputs are being replaced by a `dataclass`, so they'll be changed soon enough (no loss in backwards compatibility).
- The `base_model_prefix`es have been updated to correctly reference the base model. This is important for the `get_input_embeddings` and `set_input_embeddings` methods.

### CODE REVIEW

File: src/transformers/modeling_dpr.py [hunk 1]
```diff
@@ -256,16 +292,49 @@ def __init__(self, config: DPRConfig):
...
+            input_ids=input_ids,
+            attention_mask=attention_mask,
+            token_type_ids=token_type_ids,
+            inputs_embeds=inputs_embeds,
+            output_attentions=output_attentions,
+            output_hidden_states=output_hidden_states,
         )
-        return pooled_output
+        sequence_output, pooled_output = outputs[:2]
+        return (pooled_output,) + outputs[2:]
```
Comment (LysandreJik) [inline src/transformers/modeling_dpr.py:337]: This is now a tuple output

File: src/transformers/modeling_dpr.py [hunk 2]
```diff
@@ -298,16 +367,45 @@ def __init__(self, config: DPRConfig):
...
```
Comment (LysandreJik) [inline src/transformers/modeling_dpr.py:412]: This is now a tuple output

File: tests/test_modeling_dpr.py [hunk 3]
```diff
@@ -190,10 +186,15 @@ def prepare_config_and_inputs_for_common(self):
 
 
 @require_torch
-class DPRModelTest(unittest.TestCase):
+class DPRModelTest(ModelTesterMixin, unittest.TestCase):
 
     all_model_classes = (DPRContextEncoder, DPRQuestionEncoder, DPRReader,) if is_torch_available() else ()
 
+    test_resize_embeddings = False
+    test_missing_keys = False  # why?
```
Comment (lhoestq) [inline tests/test_modeling_dpr.py:194]: Why "why" ?

**Review by lhoestq — APPROVED** [review]
Awesome thanks !
